src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -7398,11 +7398,23 @@ public static void InternalTestAotSubset()
             Assert.True(s.IsNormalized(NormalizationForm.FormD));
             Assert.True(s.IsNormalized(NormalizationForm.FormKC));
             Assert.True(s.IsNormalized(NormalizationForm.FormKD));
-            Assert.Same(s, s.Normalize());
-            Assert.Same(s, s.Normalize(NormalizationForm.FormC));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormD));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormKC));
-            Assert.Same(s, s.Normalize(NormalizationForm.FormKD));
+
+            if (PlatformDetection.IsMonoRuntime)
```

Comment:
Would it be better to fix this in Mono instead?

Reply:
Not trivial, at least not with the same performance constraints. CoreCLR has the special `IsAscii` method that caches the value in object header. It then uses it to optimize this special case. The code is in the shared CoreLib part but it's guarded with `#if CORECLR` so this specific optimization is not used by any other runtime now. The implementation of `IsAscii` would be trivial (ASCIIUtility class even has a vectorized version), but the caching may be tricky to implement.

Reply:
The important part is avoiding the PInvoke and allocation. I do not think caching `IsAscii` is important. Caching `IsAscii` helps for microbenchmarks that runs the function against the same string in a loop; but it tend to hurt in real code that runs the function against unique string instances.

I would be fine with getting rid of the ifdef, and calling ASCIIUtility for all runtimes. It would be an improvement across the board.

Reply:
Great. I will submit a PR for that.

Reply:
Ok so I guess I can remove my change since ASCIIUtility will help mono too